src/syft/lib/python/range.py

```diff
@@ -0,0 +1,127 @@
+# stdlib
+from typing import Any
+from typing import Optional
+from typing import Union
+
+# third party
+from google.protobuf.reflection import GeneratedProtocolMessageType
+
+# syft relative
+from ... import deserialize
+from ...core.common import UID
+from ...core.common.serde.serializable import bind_protobuf
+from ...proto.lib.python.range_pb2 import Range as Range_PB
+from .iterator import Iterator
+from .primitive_factory import PrimitiveFactory
+from .primitive_interface import PyPrimitive
+from .types import SyPrimitiveRet
+
+
+@bind_protobuf
+class Range(PyPrimitive):
+    __slots__ = ["_id", "_index"]
+
+    def __init__(
+        self,
+        start: Any = None,
+        stop: Union[Any] = None,
+        step: Union[Any] = 1,
+        id: Optional[UID] = None,
+    ):
+        if stop is None:
+            stop = start
+            start = 0
+        self.value = range(start, stop, step)
+        self._id: UID = id if id else UID()
+
+    @property
+    def id(self) -> UID:
+        """We reveal PyPrimitive.id as a property to discourage users and
+        developers of Syft from modifying .id attributes after an object
+        has been initialized.
+
+        :return: returns the unique id of the object
+        :rtype: UID
+        """
+        return self._id
+
+    def __contains__(self, other: Any) -> SyPrimitiveRet:
+        res = self.value.__contains__(other)
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    def __eq__(self, other: Any) -> SyPrimitiveRet:
+        res = self.value.__eq__(other)
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    def __ne__(self, other: Any) -> SyPrimitiveRet:
+        res = self.value.__ne__(other)
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    def __sizeof__(self) -> SyPrimitiveRet:
+        res = self.value.__sizeof__()
+        return PrimitiveFactory.generate_primitive(value=res)
+
+    def __bool__(self) -> SyPrimitiveRet:
+        res = self.value.__bool__()
+        return PrimitiveFactory.generate_primitive(value=res)
```

Comment:
This should return __bool__ object, not pointer (as in https://github.com/OpenMined/PySyft/pull/5452/commits/5ce1dd63b5246be7ec94f38af91e8815f8906ac6).

Reply:
```python
from syft.lib.python.range import Range
print(type(Range(2).__bool__()))
```
this prints 
`<class 'syft.lib.python.Bool'>`

So, this is returning a PySyft bool object.